packages/wonder-blocks-link/src/components/link-core.tsx

```diff
@@ -79,15 +82,38 @@ export default class LinkCore extends React.Component<Props> {
             <Icon
                 icon={externalIconPath}
                 size="small"
-                style={iconStyles.icon}
+                style={iconStyles.endIcon}
                 testId="external-icon"
             />
         );
 
+        const startIconChild = startIcon && (
+            <Icon
+                icon={startIcon}
+                size="small"
+                style={iconStyles.startIcon}
+                testId="start-icon"
+                aria-hidden="true"
+            />
+        );
+
+        const endIconChild = endIcon ? (
+            <Icon
+                icon={endIcon}
+                size="small"
+                style={iconStyles.endIcon}
+                testId="end-icon"
+                aria-hidden="true"
+            />
+        ) : (
+            target === "_blank" && externalIcon
+        );
+
         const linkContent = (
             <>
-                {children}
-                {target === "_blank" && externalIcon}
+                {startIconChild}
```

Comment:
I think the convention would be to remove the `startIconChild &&` from the const definition (for both start icon and end icon), and add it down here.

```
const startIconChild = <Icon />

const linkContent = (
    <>
         {startIcon && startIconChild}
         ...
     </>
 )
```

Reply:
When I add `as IconAsset` the error goes away, but I'm not sure if this goes against the point of using Typescript.

<img width="300" alt="Screenshot 2023-04-11 at 1 40 41 PM" src="https://user-images.githubusercontent.com/100858764/231245030-fc58ba54-1b41-42b1-ac38-8d77aeb472f1.png">

Reply:
^ I'm going to go with the last implementation because I see it done for other components and it passes the typecheck! 😊